### PR TITLE
Refactor(eos_designs): Setting ospf.area default values to 0.0.0.0

### LINKED ISSUE #3006: Feat(eos_designs): OSFP default values should default to `"0.0.0.0"` since that is how EOS translates `"0"`
OSFP default values should default to `"0.0.0.0"` since that is how EOS translates `"0"`

_Originally posted by @ClausHolbechArista in https://github.com/aristanetworks/ansible-avd/issues/2998#issuecomment-1601348342_

### PR DESCRIPTION
**Pull request by Shivani-gslab**
## Change Summary

OSPF default values should default to "0.0.0.0" since that is how EOS translates "0"

## Related Issue(s)

Fixes #3006 

## Component(s) name

`arista.avd.eos_designs`

## Proposed changes

Setting ospf.area default values to 0.0.0.0 in schema and code.

### User Checklist

- N/A

### Repository Checklist

- [x] My code has been rebased from devel before I start
- [x] I have read the [**CONTRIBUTING**](https://avd.sh/en/latest/docs/contribution/overview.html) document.
- [x] My change requires a change to the documentation and documentation have been updated accordingly.
- [x] I have updated [molecule CI](https://github.com/aristanetworks/avd/tree/devel/ansible_collections/arista/avd/molecule) testing accordingly. (check the box if not applicable)

### CODE REVIEW

**Comment by gmuloc** [comment]
@Shivani-gslab we have a couple of places in our molecule scenarios where the value was hardcoded to `0` for area under tenants:

```
./eos_designs-mpls-isis-sr-ldp/inventory/group_vars/MPLS_TENANTS_NETWORKS.yml:              area: 0
./eos_designs-mpls-isis-sr-ldp/inventory/group_vars/MPLS_TENANTS_NETWORKS.yml:              area: 0
./eos_designs_unit_tests/inventory/group_vars/DC1_TENANTS_NETWORKS/Tenant_A.yml:              area: 0
./eos_designs_unit_tests/inventory/group_vars/DC1_TENANTS_NETWORKS/Tenant_A.yml:              area: 0
./eos_designs_unit_tests/inventory/group_vars/DC1_TENANTS_NETWORKS/Tenant_A.yml:              area: 0
```

I think for the sake of them being example it would be best to update them to 0.0.0.0 in this PR.

File: ansible_collections/arista/avd/docs/porting-guides/5.x.x.md [hunk 1]
```diff
@@ -276,6 +276,24 @@ To retain the old configuration with no TCAM profile the default platform settin
 +      non_mlag: 1020
 ```
 
+### Default value for OSPF area has been set to `0.0.0.0`
+
+With AVD version 5.0.0, the default value for OSPF area has been set to "0.0.0.0" to align with how EOS interprets the value "0".
```
Comment (Vibhu-gslab) [inline ansible_collections/arista/avd/docs/porting-guides/5.x.x.md]:
```suggestion
With AVD version 5.0.0, the default value for OSPF area has been updated to "0.0.0.0" to align with EOS's interpretation of the value "0".
```

**Review by Vibhu-gslab — APPROVED** [review]

File: ansible_collections/arista/avd/docs/porting-guides/5.x.x.md [hunk 2]
```diff
@@ -276,6 +276,24 @@ To retain the old configuration with no TCAM profile the default platform settin
 +      non_mlag: 1020
 ```
 
+### Default value for OSPF area has been set to `0.0.0.0`
+
```
Comment (laxmikantchintakindi) [inline ansible_collections/arista/avd/docs/porting-guides/5.x.x.md:280]: We may not need to mention this in porting guide.

File: ansible_collections/arista/avd/docs/release-notes/5.x.x.md [hunk 3]
```diff
@@ -160,6 +160,10 @@ Starting AVD 5.0.0, the default MTU for Dps1, WAN HA and WAN p2p uplink interfac
 
 See the [porting guide](../porting-guides/5.x.x.md#default-mtu-value-changed-from-9214-to-9194-for-dps1-wan-ha-and-wan-p2p-uplink-interfaces) for details.
 
+#### Default value for OSPF area has been set to "0.0.0.0"
```
Comment (Shivani-gslab) [inline ansible_collections/arista/avd/docs/release-notes/5.x.x.md]:
```suggestion
#### Default value for OSPF area has been updated to "0.0.0.0"
```

File: ansible_collections/arista/avd/docs/porting-guides/5.x.x.md [hunk 4]
```diff
@@ -276,6 +276,24 @@ To retain the old configuration with no TCAM profile the default platform settin
 +      non_mlag: 1020
 ```
 
+### Default value for OSPF area has been set to `0.0.0.0`
```
Comment (Shivani-gslab) [inline ansible_collections/arista/avd/docs/porting-guides/5.x.x.md]:
```suggestion
### Default value for OSPF area has been updated to `0.0.0.0`
```

File: ansible_collections/arista/avd/docs/release-notes/5.x.x.md [hunk 5]
```diff
@@ -160,6 +160,10 @@ Starting AVD 5.0.0, the default MTU for Dps1, WAN HA and WAN p2p uplink interfac
...
+
+See the [porting guide](../porting-guides/5.x.x.md#default-value-for-ospf-area-has-been-set-to-0000) for details.
```
Comment (Shivani-gslab) [inline ansible_collections/arista/avd/docs/release-notes/5.x.x.md]:
```suggestion
See the [porting guide](../porting-guides/5.x.x.md#default-value-for-ospf-area-has-been-updated-to-0000) for details.
```

File: ansible_collections/arista/avd/molecule/eos_designs_unit_tests/intended/configs/DC1-LEAF2B.cfg
Comment (alexeygorbunov) [inline ansible_collections/arista/avd/molecule/eos_designs_unit_tests/intended/configs/DC1-LEAF2B.cfg:1]: Line 303 contains reference to area 0:
```
   ip ospf area 0
```

File: ansible_collections/arista/avd/molecule/eos_designs_unit_tests/intended/structured_configs/DC1-LEAF2B.yml
Comment (alexeygorbunov) [inline ansible_collections/arista/avd/molecule/eos_designs_unit_tests/intended/structured_configs/DC1-LEAF2B.yml:1]: line 628
```
  ospf_area: '0'
```

**Review by alexeygorbunov — COMMENTED** [review]
`./ansible_collections/arista/avd/molecule/eos_designs-mpls-isis-sr-ldp/documentation/devices/SITE1-LER1.md`

285:   ip ospf area 0
600:| Ethernet6.100 | 0 | 10 | True |

File: ansible_collections/arista/avd/molecule/eos_designs_unit_tests/intended/configs/DC1-LEAF2B.cfg
Comment (Shivani-gslab) [reply ansible_collections/arista/avd/molecule/eos_designs_unit_tests/intended/configs/DC1-LEAF2B.cfg:1]: This is because it is provided as **input** here and not generated by default -https://github.com/aristanetworks/avd/blob/devel/ansible_collections/arista/avd/molecule/eos_designs_unit_tests/inventory/group_vars/DC1_TENANTS_NETWORKS/Tenant_A.yml#L287

File: ansible_collections/arista/avd/molecule/eos_designs_unit_tests/intended/structured_configs/DC1-LEAF2B.yml
Comment (Shivani-gslab) [reply ansible_collections/arista/avd/molecule/eos_designs_unit_tests/intended/structured_configs/DC1-LEAF2B.yml:1]: https://github.com/aristanetworks/avd/blob/devel/ansible_collections/arista/avd/molecule/eos_designs_unit_tests/inventory/group_vars/DC1_TENANTS_NETWORKS/Tenant_A.yml#L287

**Comment by Shivani-gslab** [comment]
> `./ansible_collections/arista/avd/molecule/eos_designs-mpls-isis-sr-ldp/documentation/devices/SITE1-LER1.md`
> 
> 285: ip ospf area 0 600:| Ethernet6.100 | 0 | 10 | True |

it is provided as input here and is not generated by default - https://github.com/aristanetworks/avd/blob/devel/ansible_collections/arista/avd/molecule/eos_designs-mpls-isis-sr-ldp/inventory/group_vars/MPLS_TENANTS_NETWORKS.yml#L79

File: ansible_collections/arista/avd/docs/porting-guides/5.x.x.md (hunk 2 above)
Comment (carlbuchmann) [reply ansible_collections/arista/avd/docs/porting-guides/5.x.x.md:280]: Yes, we should include details on how to revert the default. I will take care of it.

**Comment by ClausHolbechArista** [comment]
@gmuloc @alexeygorbunov To be clear, we still need to test that giving ospf area 0 still works. So it is ok to have molecule vars with area 0. It is just the default we are changing.

**Comment by ClausHolbechArista** [comment]
Guillaume told me we have some test with another integer area, so that we can keep that and then change all the 0 -> 0.0.0.0 in molecule inputs.

File: ansible_collections/arista/avd/molecule/eos_designs_unit_tests/inventory/group_vars/DC1_TENANTS_NETWORKS/Tenant_A.yml [hunk 6]
```diff
@@ -45,6 +45,7 @@ tenant_a:
             enabled: True
             ospf:
               enabled: true
+              # Keeping 0 to make sure using 0 still works
```
Comment (gmuloc) [inline ansible_collections/arista/avd/molecule/eos_designs_unit_tests/inventory/group_vars/DC1_TENANTS_NETWORKS/Tenant_A.yml]:
```suggestion
              # Keeping to make sure using '0' still works
```

File: ansible_collections/arista/avd/docs/porting-guides/5.x.x.md [hunk 7]
```diff
@@ -276,6 +276,24 @@ To retain the old configuration with no TCAM profile the default platform settin
...
+    peer_type: l3_interface
+    ip_address: 10.0.1.1/29
+    shutdown: false
+    description: C1_L3_SERVICE
+    encapsulation_dot1q:
+      vlan: 10
+    vrf: C1_VRF1
+-    ospf_area: '0'
++    ospf_area: 0.0.0.0
+```
```
Comment (ClausHolbechArista) [inline ansible_collections/arista/avd/docs/porting-guides/5.x.x.md]: This should show how to revert the change by setting the various vars where we have changed the default.

**Review by ClausHolbechArista — DISMISSED** [review]

File: ansible_collections/arista/avd/docs/porting-guides/5.x.x.md (hunk 7 above)
Comment (Shivani-gslab) [reply ansible_collections/arista/avd/docs/porting-guides/5.x.x.md]: Fixed

File: ansible_collections/arista/avd/docs/porting-guides/5.x.x.md [hunk 8]
```diff
@@ -276,6 +276,53 @@ To retain the old configuration with no TCAM profile the default platform settin
...
+            enabled: true
+            interfaces:
+              - Ethernet6.10
+            ip_addresses:
+              - 123.1.1.0/31
+            nodes:
+              - SITE1-LER1
+            ospf:
+              enabled: true
++              area: 0
```
Comment (laxmikantchintakindi) [inline ansible_collections/arista/avd/docs/porting-guides/5.x.x.md]:
```suggestion
+             area: 0
```

File: ansible_collections/arista/avd/docs/porting-guides/5.x.x.md [hunk 9]
```diff
@@ -276,6 +276,53 @@ To retain the old configuration with no TCAM profile the default platform settin
...
+              cost: 10
+        loopbacks:
+          - loopback: 123
+            ip_address: 10.1.53.0/32
+            node: DC1-LEAF2A
+            description: Test-Loopback
+            enabled: true
+            ospf:
+              enabled: true
++              area: 0
```
Comment (laxmikantchintakindi) [inline ansible_collections/arista/avd/docs/porting-guides/5.x.x.md]:
```suggestion
+             area: 0
```

File: ansible_collections/arista/avd/docs/porting-guides/5.x.x.md [hunk 10]
```diff
@@ -276,6 +276,53 @@ To retain the old configuration with no TCAM profile the default platform settin
...
+        svis:
+          - id: 113
+            name: Tenant_A_OP_Zone_4
+            tags: [ DC1_LEAF1 ]
+            enabled: True
+            ospf:
+              enabled: true
++              area: 0
```
Comment (laxmikantchintakindi) [inline ansible_collections/arista/avd/docs/porting-guides/5.x.x.md]:
```suggestion
+             area: 0
```

**Review by gmuloc — APPROVED** [review]

**Review by carlbuchmann — APPROVED** [review]
LGTM
